lib/Mojo/Base.pm

```diff
@@ -15,6 +15,10 @@ use IO::Handle ();
 my $NAME
   = eval { require Sub::Util; Sub::Util->can('set_subname') } || sub { $_[1] };
 
+# Role support requires Role::Tiny 2.000001+
+use constant ROLES =>
+  !!(eval 'require Role::Tiny' && $Role::Tiny::VERSION >= 2.000001);
```

Comment:
Actually the best method would be `eval { require Role::Tiny; Role::Tiny->VERSION('2.000001'); 1 }`

String eval is unnecessary for this, and the ->VERSION method (same as used by use) is the correct way to check a module's version.